doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -774,10 +777,7 @@ in a less hard-wired way.
       I don't when which is required
 
    NOT RIGHT YET: In the :token:`ltac2_match_pattern`\s, metavariables have the form :n:`?@ident`, whereas
-   in the :n:`@ltac2_expr`\s, the question mark is omitted.  Choose your metavariable
-   names with care to avoid name conflicts.  For example, if you use the metavariable `S`,
-   then the :token:`ltac2_expr` can't use `S` to refer to the constructor of `nat`
-   without qualifying the constructor as `Datatypes.S`.
+   in the :n:`@ltac2_expr`\s, the question mark is omitted.
```

Comment:
This seems right now, no?

Reply:
Yeah.  If the right-hand side refers to a global symbol `x`, then it would be a user error to choose `x` as a metavariable.  And presumably if `x` is a bound variable in the RHS that will be handled correctly (not substituted).